### PR TITLE
Expose CoreDNS over tcp + make Dig in terratests go via tcp

### PR DESCRIPTION
**Pull request by jkremser**
why: When we get rid of Docker Desktop and use for instance Lima, it doesn't have support for UDP port forwarding from host to containers running in the VM - https://github.com/lima-vm/lima/issues/366

So as a workaround, we can run the Dig in our tests using the tcp protocol, where the redirect works fine (it uses ssh tunnel). 

~tests will fail, it assumes a new release on `AbsaOSS/gopkg` with this https://github.com/AbsaOSS/gopkg/pull/11 merged in~

It adds a new service for coredns that uses NodePort (30053) and then adds this redirect in k3d manifests so that

```
dig @127.0.0.1 -p5053 failover.cloud.example.com +short +tcp
dig @127.0.0.1 -p5054 failover.cloud.example.com +short +tcp
```
both work

~todo: update docs for local playground~

that `AbsaOSS/gopkg` library contains a module called strings, that was renamed to string like a year ago, but only now it's [failing](https://github.com/k8gb-io/k8gb/runs/5069047324?check_suite_focus=true#step:6:7) because of the newly released version. Hence the `strings -> string` thing.

btw:
> DNS resolvers and recursive servers MUST support UDP, and SHOULD support TCP, for sending (non-zone-transfer) queries.
 --- https://datatracker.ietf.org/doc/html/rfc7766

Signed-off-by: Jirka Kremser <jiri.kremser@gmail.com>

### CODE REVIEW

**Review by ytsarev — COMMENTED** [review]
Aren't we making here a strong assumption that k8gb contributor should use mac+lima?

File: chart/k8gb/templates/coredns-tcp-svc.yaml [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+  name: {{ .Release.Name }}-coredns-tcp
+spec:
+  type: NodePort
+  selector:
+    app.kubernetes.io/instance: k8gb
+    app.kubernetes.io/name: coredns
+  ports:
+  - name: tcp-5353
+    port: 53
+    protocol: TCP
```
Comment (ytsarev) [inline chart/k8gb/templates/coredns-tcp-svc.yaml:16]: Was it tested outside of the local scenario? Main concern is overall TCP support/expectation in the DNS environment

**Review by somaritane — APPROVED** [review]
lgtm

**Review by kuritka — APPROVED** [review]

File: chart/k8gb/templates/coredns-tcp-svc.yaml (hunk 1 above)
Comment (k0da) [reply chart/k8gb/templates/coredns-tcp-svc.yaml:16]: @jkremser could you please move this service definition out of helm and deploy it during test setup creation (around we deploy edgeDNS)

**Review by k0da — CHANGES_REQUESTED** [review]

File: chart/k8gb/templates/coredns-tcp-svc.yaml (hunk 1 above)
Comment (jkremser) [reply chart/k8gb/templates/coredns-tcp-svc.yaml]: it should be there

**Comment by k0da** [comment]
This way it will be deployed into default namespace

**Comment by jkremser** [comment]
> This way it will be deployed into default namespace

good point, added `k8gb` ns

**Review by k0da — DISMISSED** [review]
LGTM

**Review by k0da — APPROVED** [review]

**Pull request merged**